drivers/adc/max9611pmb/max9611.c

```diff
@@ -0,0 +1,465 @@
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include <stdint.h>
+#include <stdlib.h>
+#include "no_os_i2c.h"
+#include "no_os_util.h"
+#include "no_os_error.h"
+#include "no_os_delay.h"
+#include "max9611.h"
+
+/***************************************************************************//**
+ * @brief Generates slave address based on A1 and A0 pin values
+ * @param a0 - A0 logic level
+ * @param a1 - A1 logic level
+ * @return addr - Computed slave ID
+*******************************************************************************/
+int max9611_addr_gen(enum max9611_addr_lvls a0, enum max9611_addr_lvls a1)
+{
+	int addr;
+
+	switch (a1) {
+	case MAX9611_ZERO_VCC:
+		addr = 0xE0;
+		break;
+	case MAX9611_33_VCC:
+		addr = 0xE8;
+		break;
+	case MAX9611_66_VCC:
+		addr = 0xF0;
+		break;
+	case MAX9611_100_VCC:
+		addr = 0xF8;
+		break;
+	default:
+		addr = 0xE0;
+		break;
+	}
+
+	addr += (2 * a0);
+
+	return addr >> 1;
+}
+
+/***************************************************************************//**
+ * @brief Initializes the communication peripheral and checks if the device is
+ *        present.
+ *
+ * @param device     - The device structure.
+ * @param init_param - The structure that contains the device initial
+ * 		       parameters.
+ *
+ * @return ret - The result of the initialization procedure.
+ *                  Example: -1 - I2C peripheral wasn't initialized or the
+ *                                device is not present.
+ *                            0 - I2C peripheral was initialized and the
+ *                                device is present.
+*******************************************************************************/
+int max9611_init(struct max9611_dev **device,
+		 struct max9611_init_param init_param)
+{
+	int ret;
+	struct max9611_dev *dev;
+
+	dev = (struct max9611_dev *)calloc(1, sizeof(*dev));
```

Comment:
please see the latest no_os_calloc and no_os_malloc and no_os_free

Reply:
Utilized in latest commit.